src/exchange-connector/exchangeDirectoryParser.ts

```diff
@@ -52,8 +53,7 @@ export function extractFiles(
           );
         })
       );
-    }
-  });
+    });
 
   // The then here collapses the return from an array of void to a single void
```

Comment:
Minor issue, but [this is not true](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then).